### PR TITLE
Adds tests to GenerateDotStream().

### PR DESCRIPTION
**Pull request by agalbachicar**
# 🎉 New feature

Complements https://github.com/maliput/maliput/pull/637 and https://github.com/maliput/maliput/pull/639

## Summary
Provides test for the DOT serialization of graphs and routes.

## Test it

Via colcon test

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/generate_dot_test.cc [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+4 -- 5 [ label = "2_0" ];
+2 -- 5 [ label = "6_0" ];
+1 -- 2 [ label = "4_0" ];
+2 -- 3 [ label = "1_0" ];
+0 -- 1 [ label = "0_0" color = "red" ];
+}
+)");
+
+  std::stringstream ss;
+  maliput::utility::GenerateDotStream(graph_, routes_.front(), &ss);
```
Comment (liangfok) [inline test/generate_dot_test.cc:113]: > maliput::

Given that this is already in the "maliput" namespace, is this necessary? Ditto below.

File: test/generate_dot_test.cc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                                  api::LanePosition(1., 0., 0.));
+    const api::RoadPosition end(road_network_->road_geometry()->ById().GetLane(kEndLaneId),
+                                api::LanePosition(10., 0., 0.));
+    routes_ = router_->ComputeRoutes(start, end, kDefaultRoutingConstraints);
+    ASSERT_EQ(1u, routes_.size());
+  }
+};
+
+// Evaluates the graph is correct and the api::Segment containing the routing::Route is marked with the red color.
+TEST_F(TShapeRoadGenerateDotTest, DotGraphWithARouteMarksInRedWhereTheRouteGoesThrough) {
```
Comment (liangfok) [inline test/generate_dot_test.cc:98]: > DotGraphWithARouteMarksInRedWhereTheRouteGoesThrough

Consider simplifying to be "DotGraphWithRouteHighlightedInRed."

File: test/generate_dot_test.cc [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+)");
+
+  std::stringstream ss;
+  maliput::utility::GenerateDotStream(graph_, routes_.front(), &ss);
+
+  ASSERT_EQ(kResult, ss.str());
+}
+
+// Evaluates the graph serialization is the expected one. There is no prescriptive order in the serialization, thus
+// differences are expected in the order of the edges.
```
Comment (liangfok) [inline test/generate_dot_test.cc:119]: > There is no prescriptive order in the serialization, thus differences are expected in the order of the edges.

Does the code below handle randomly-ordered edges?

**Review by liangfok — CHANGES_REQUESTED** [review]
Completed a pass.

File: test/generate_dot_test.cc (hunk 1 above)
Comment (agalbachicar) [reply test/generate_dot_test.cc]: [db8afbd](https://github.com/maliput/maliput_integration_tests/pull/85/commits/db8afbd2ce9695f2fe50d579c4132901bcfaf4ea)

File: test/generate_dot_test.cc (hunk 2 above)
Comment (agalbachicar) [reply test/generate_dot_test.cc]: [db8afbd](https://github.com/maliput/maliput_integration_tests/pull/85/commits/db8afbd2ce9695f2fe50d579c4132901bcfaf4ea)

File: test/generate_dot_test.cc (hunk 3 above)
Comment (agalbachicar) [reply test/generate_dot_test.cc:148]: [db8afbd](https://github.com/maliput/maliput_integration_tests/pull/85/commits/db8afbd2ce9695f2fe50d579c4132901bcfaf4ea)

**Review by liangfok — APPROVED** [review]
LGTM

**Pull request merged**